### PR TITLE
feat: add support for radxa rock 4c+

### PR DESCRIPTION
**Pull request by DamiaPoquet**
Adding support for RK3399-based Radxa ROCK 4c+ board.

### CODE REVIEW

File: Pkgfile [hunk 1]
```diff
@@ -9,8 +9,8 @@ vars:
...
   # renovate: datasource=github-tags extractVersion=^v(?<version>.*)$ depName=u-boot/u-boot
-  uboot_version: 2023.01
-  uboot_sha256: 69423bad380f89a0916636e89e6dcbd2e4512d584308d922d1039d1e4331950f
-  uboot_sha512: 417a28267eb7875820d08fafc7316f164663609378637539e71648b0b9b7d28796b6c381717f31b0ab6472805fefd32628ef7d1b2e7b9f3c51c8ad122993f679
+  uboot_version: 2024.01
+  uboot_sha256: b99611f1ed237bf3541bdc8434b68c96a6e05967061f992443cb30aabebef5b3
+  uboot_sha512: 45bd093ba3bda23e43cdde83d8656c1ee1348ac2886ecff1fee475f101ac4965a5be6565408fa5b990c723f3fdc833edfca60a719f735a43040cd14a1b59a88b
 labels:
-  org.opencontainers.image.source: https://github.com/siderolabs/sbc-raspberrypi
+  org.opencontainers.image.source: https://github.com/siderolabs/sbc-rockchip
```
Comment (frezbo) [inline Pkgfile:16]: good catch

File: Pkgfile [hunk 2]
```diff
@@ -9,8 +9,8 @@ vars:
   arm_trusted_firmware_sha512: 958cf8f9e258638e59d0fbd8b053fce7d8a9ea2fc922686c9d20ea16f79f55219ac18a12ab240c528ee98e49c2e0eef4c963fdb255cc14b92437a5b3cffc8640
 
   # renovate: datasource=github-tags extractVersion=^v(?<version>.*)$ depName=u-boot/u-boot
-  uboot_version: 2023.01
-  uboot_sha256: 69423bad380f89a0916636e89e6dcbd2e4512d584308d922d1039d1e4331950f
-  uboot_sha512: 417a28267eb7875820d08fafc7316f164663609378637539e71648b0b9b7d28796b6c381717f31b0ab6472805fefd32628ef7d1b2e7b9f3c51c8ad122993f679
+  uboot_version: 2024.01
```
Comment (frezbo) [inline Pkgfile:12]: i think this will break the other builds since it has patches

Comment (DamiaPoquet) [reply Pkgfile:12]: I'm reviewing this point and I'll get back to you

File: installers/rock4cplus/src/go.mod [hunk 3]
```diff
@@ -0,0 +1,11 @@
+module rockpi4c
```
Comment (frezbo) [inline installers/rock4cplus/src/go.mod:1]:
```suggestion
module rockpi4cplus
```

File: Pkgfile (hunk 2 above)
Comment (DamiaPoquet) [reply Pkgfile:12]: Yes, the only patches are for enabling the SPI chip support. Most installs happen on SD cards or eMMC modules; not all boards are sold with pre-soldered SPI chips. 

Dropping support for the SPI flash build will ease the process while impacting few users.

Comment (frezbo) [reply Pkgfile:12]: I was thinking of dropping it in a different PR and clean up the docs, I don't mind dropping here also

Comment (frezbo) [reply Pkgfile:12]: I've dropped all patches here https://github.com/siderolabs/sbc-rockchip/pull/6

Comment (DamiaPoquet) [reply Pkgfile:12]: Perfect! I'll rebase this branch after merging the new PR. Thanks!

**Review by frezbo — APPROVED** [review]

**Pull request merged**